app/containers/LoginServices.tsx

```diff
@@ -92,154 +94,145 @@ interface IOauthProvider {
 	wordpress: () => void;
 }
 
-interface ILoginServicesProps {
-	navigation: StackNavigationProp<OutsideParamList>;
-	server: string;
-	services: IServices;
-	Gitlab_URL: string;
-	CAS_enabled: boolean;
-	CAS_login_url: string;
-	separator: boolean;
-	theme: TSupportedThemes;
-}
-
-interface ILoginServicesState {
-	collapsed: boolean;
-	servicesHeight: Animated.Value;
-}
-
-class LoginServices extends React.PureComponent<ILoginServicesProps, ILoginServicesState> {
-	private _animation?: Animated.CompositeAnimation | void;
-
-	state = {
-		collapsed: true,
-		servicesHeight: new Animated.Value(SERVICES_COLLAPSED_HEIGHT)
-	};
-
-	onPressFacebook = () => {
+const LoginServices = (): React.ReactElement => {
+	const [collapsed, setCollapsed] = useState(true);
+
+	const navigation = useNavigation<StackNavigationProp<OutsideParamList>>();
+	const { theme } = useTheme();
+
+	const { Gitlab_URL, CAS_enabled, CAS_login_url, Accounts_ShowFormLogin } = useAppSelector(
+		state => ({
+			Gitlab_URL: state.settings.API_Gitlab_URL as string,
+			CAS_enabled: state.settings.CAS_enabled as boolean,
+			CAS_login_url: state.settings.CAS_login_url as string,
+			Accounts_ShowFormLogin: state.settings.Accounts_ShowFormLogin as boolean
+		}),
+		shallowEqual
+	);
+	const server = useAppSelector(state => state.server.server);
+	const services = useAppSelector(state => state.login.services as IServices);
+	const { length } = Object.values(services);
+
+	const heightButtons = useSharedValue(SERVICES_COLLAPSED_HEIGHT);
+
+	const animatedStyle = useAnimatedStyle(() => ({
+		overflow: 'hidden',
+		height: withTiming(heightButtons.value, { duration: 300, easing: Easing.inOut(Easing.quad) })
+	}));
+
+	const onPressFacebook = () => {
```

Comment:
On our refinement, we talked about extracting all these service logic functions away from the component.